connectors/sources/confluence.py

```diff
@@ -210,6 +212,34 @@ async def paginated_api_call(self, url_name, **url_kwargs):
                 )
                 break
 
+    async def paginated_api_call_for_datacenter_syncrule(self, url_name, **url_kwargs):
+        """Make a paginated API call for datacenter using the passed url_name.
+        Args:
+            url_name (str): URL Name to identify the API endpoint to hit
+        Yields:
+            response: JSON response.
+        """
+        while True:
+            url = os.path.join(self.host_url, URLS[url_name].format(**url_kwargs))
+            json_response = {}
+            try:
+                async for response in self.api_call(
+                    url=url,
+                ):
+                    json_response = await response.json()
+                    yield json_response
+
+                    start = url_kwargs.get("start", 0)
+                    start += 100
+                    url_kwargs["start"] = start
+                if len(json_response.get("results", [])) < 100:
+                    break
+            except Exception as exception:
+                self._logger.warning(
+                    f"Skipping data for type {url_name} from {url}. Exception: {exception}."
+                )
+                break
```

Comment:
Why not reraise here? It can make it look like sync is successful, but actually fail

Reply:
If I raise it here then sync will be stop here and not work for other queries, issues etc. We are raising the error after retrying in api_call method

Reply:
What if every resource raises an error, will it result in successful sync?

> We are raising the error after retrying in api_call method
But I see retry relies on the method raising an error, so if you don't raise you don't retry?

Reply:
I guess this is the worst case that every resource raise error, even in such condition we'll skip all the resources and show successful sync message. We are following same approach for every connector right now. 
Also if api call throw any error in that case only we'll retry 3 times and raise the error with failure message on the terminal.